src/mon/OSDMonitor.cc

```diff
@@ -395,14 +405,22 @@ void OSDMonitor::update_from_paxos(bool *need_bootstrap)
       t->erase("mkfs", "osdmap");
     }
 
+    // make sure we're using the right pg service.. remove me post-luminous!
+    if (osdmap.require_osd_release >= CEPH_RELEASE_LUMINOUS) {
+      dout(10) << __func__ << " pgservice is mgrstat" << dendl;
+      mon->pgservice = mon->mgrstatmon()->get_pg_stat_service();
+    } else {
+      dout(10) << __func__ << " pgservice is pg" << dendl;
+      mon->pgservice = mon->pgmon()->get_pg_stat_service();
+    }
+
     if (tx_size > g_conf->mon_sync_max_payload_size*2) {
       mon->store->apply_transaction(t);
       t = MonitorDBStore::TransactionRef();
       tx_size = 0;
     }
     if (mon->monmap->get_required_features().contains_all(
           ceph::features::mon::FEATURE_LUMINOUS)) {
-      creating_pgs = update_pending_pgs(inc);
```

Comment:
@tchaikov @liewegas did this ever get discussed? From commit:

> We are not persisiting the updated creating_pgs here; this is bad!  I'm not sure why it was there to begin with?

It originated in 43ded67243c88a1abb7f07aa4dbdc326eec7633a as its own hunk, so it was clearly deliberate. I think it must have been so we could identify some of the state-changed PGs and I'm a bit worried about dropping the hunk. (For instance, if we've deleted a pool with creating pgs, we don't want to tell OSDs to create them! And we don't need to persist that data since any subsequent load will produce the same outcome.)